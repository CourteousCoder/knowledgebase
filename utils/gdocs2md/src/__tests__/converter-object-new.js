const fs = require("fs");
const path = require(`path`);

const { ElementsOfGoogleDocument } = require("../elements-of-google-document");

const documentsPath = path.join(__dirname, "gdocs-json");
const filenames = fs.readdirSync(documentsPath);
console.log("debug", documentsPath);
filenames.forEach(function (filename) {
  const filepath = path.join(documentsPath, filename);
  const file = fs.readFileSync(filepath, "utf8");
  const parsedGdoc = JSON.parse(file);
  const googleDocument = new ElementsOfGoogleDocument({
    ...parsedGdoc,
  });
  googleDocument.process();

  test(`Document "${googleDocument.document.title}" to Object`, () => {
    const { cover, elements } = googleDocument;

    expect({
      cover,
      elements,
    }).toMatchSnapshot();
  });
});
