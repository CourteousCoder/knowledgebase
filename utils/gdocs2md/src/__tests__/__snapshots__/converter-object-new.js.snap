// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document "Advanced Formatting - color and fonts" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "p",
      "value": "Times New Roman",
    },
    {
      "type": "p",
      "value": "Courier New",
    },
    {
      "type": "p",
      "value": "Impact",
    },
    {
      "type": "p",
      "value": "<span style='font-size:1.82em'>20 point font</span>",
    },
    {
      "type": "p",
      "value": "<span style='color:rgb(255, 0, 0)'>Red text</span>",
    },
    {
      "type": "p",
      "value": "<span style='color:rgb(255, 255, 255);background-color:rgb(255, 0, 255)'>White text with pink bangrounds</span>",
    },
  ],
}
`;

exports[`Document "Bold italic" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "h2",
      "value": "**Bold** and **_Bold, Italic Heading_**",
    },
    {
      "type": "p",
      "value": "This is **bold**, _italic_, **bold & italic**, <ins>underline</ins>, **_<ins>bold & italic & underline</ins>_**",
    },
  ],
}
`;

exports[`Document "Bullets and Lists" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "p",
      "value": "this is a bulleted list:",
    },
    {
      "type": "ul",
      "value": [
        "item1",
        {
          "type": "ul",
          "value": [
            "subitem 1.1",
          ],
        },
        "item 2",
        {
          "type": "ul",
          "value": [
            "item 2.1",
            "item 2.2",
            {
              "type": "ul",
              "value": [
                "item 2.2.1",
                {
                  "type": "ul",
                  "value": [
                    "item 2.2.1.1",
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
    {
      "type": "p",
      "value": "This is a numbered list",
    },
    {
      "type": "ol",
      "value": [
        "Must Have",
        {
          "type": "ol",
          "value": [
            "Feature A",
            {
              "type": "ol",
              "value": [
                "Subfeature A.1",
                "Subfeature A.2",
              ],
            },
            "Feature B",
          ],
        },
        "Should Have",
        {
          "type": "ol",
          "value": [
            "Feature X1",
            "Feature X2",
          ],
        },
        "Could Have",
        {
          "type": "ol",
          "value": [
            "Non-critical feature 1",
            "Non-critical feature 2",
            "Non-critical feature 3",
          ],
        },
        "Won't Have",
      ],
    },
  ],
}
`;

exports[`Document "Custom Frontmatter" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "p",
      "value": "---",
    },
    {
      "type": "p",
      "value": "id: 134",
    },
    {
      "type": "p",
      "value": "title: Custom Title ",
    },
    {
      "type": "p",
      "value": "category: development",
    },
    {
      "type": "p",
      "value": "---",
    },
    {
      "type": "p",
      "value": "Front matter looks like:",
    },
  ],
}
`;

exports[`Document "Footnotes" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "p",
      "value": "This sentence has detailed info in a footnote[^1]",
    },
    {
      "type": "footnote",
      "value": {
        "number": "1",
        "text": " <span style='font-size:0.91em'>This information is so tangential it needs to go into a footnote</span>",
      },
    },
  ],
}
`;

exports[`Document "Headings" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "h2",
      "value": "Most Important Topic (Heading 1)",
    },
    {
      "type": "p",
      "value": "Description of important topic",
    },
    {
      "type": "h3",
      "value": "Important Subtopic (Heading 2)",
    },
    {
      "type": "p",
      "value": "Description of subtopic",
    },
    {
      "type": "h4",
      "value": "Buried Info (Heading 3)",
    },
    {
      "type": "p",
      "value": "Description of buried info",
    },
  ],
}
`;

exports[`Document "Images" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh5.googleusercontent.com/pI7YHsq0cwDeRS2__nZNYEdzRgwyMsKPRcBBfljD7qGD0new44FBQHMp9sqeq92WHMs4BRYni11R3l4w1dYVdKmFc1tgnL4VK-YDjLRmDS15iL0iw3e6fM_Pa3zNzXZBKDwIWibc6U4dSd7ulO4qktwWjRzvZBi5Gipbf35706PqZ42_g2SRjUG3jUI4-_4HEfc",
        "title": "",
      },
    },
    {
      "type": "h2",
      "value": "The below is a small image embedded in the sentence",
    },
    {
      "type": "p",
      "value": "Here is a medium screen sho",
    },
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh5.googleusercontent.com/r1Mt25cLRPP48631g4hZ9tTLCjFDZtivODQw54INhoPVTB6IUDWebSCkm2idyskePs1E0J6fM1KZ6Z02iOEkoHYiEuOEQ7H0d6vCxewh7wfcaF6EkxEWk6FxqOGVQyr2pFEjI8qUxBGL1rN8vkRau30SYS0ym0x4UaFbo0FUWsIhG7iaRwI02oJNMDJEBP0klMA",
        "title": "",
      },
    },
    {
      "type": "h2",
      "value": "Big screenshot",
    },
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh3.googleusercontent.com/8CVxMB3Tt0sK5VvQuiqJw-vlYXzgTX2CH3i2zQsG62XgzzvitdUVDq4gdKp5zo0zp4J6azD2J6G780HZ33tOb2tS5LQ098wHB9gwee98NagPOSunLN7RirT9lfU8enFYO_htKxjUsBfko38L8YfO_AdwhfsrPSd8Ma8MwoDqJoBCxs72nUaAhFDdhNAsbyJwARM",
        "title": "",
      },
    },
    {
      "type": "p",
      "value": "t",
    },
    {
      "type": "p",
      "value": "===============================",
    },
    {
      "type": "h2",
      "value": "Image with title and description",
    },
    {
      "type": "img",
      "value": {
        "alt": "This is a wild animal.",
        "source": "https://lh3.googleusercontent.com/wRnlyznYiw5RF-IY4G83jr1c9sCYGM9w55xinE1sbxSG3f5Q4l24UFUCOIcDZES4j1mPOCRgbuVHw99yG8TX7F-87Sk0g5dVviKlS3fc3qpSJWWRdAPVFiwetsVA2h1PbSuQN59T9T0nz-tQuICI6ngLf-eidfPq2Spo0JZCN1q0ub6HVAi_9UflpAed4rK3Mjk",
        "title": "A cute but DANGEROUS cat",
      },
    },
    {
      "type": "p",
      "value": "====================================================================",
    },
    {
      "type": "h2",
      "value": "An image with a link ",
    },
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh4.googleusercontent.com/nUTGUYt2OiexwrS7UP6ebivL22RHGBiM6ReeskhJPFHDpcIdw_Piocxg5ccxQBoV-7tc1kS7J9GR7aFml0-k7u4XBwzz1S0IZpdusV_-6vkEN6uM2cCYBSNIZ0boqzaM3IAUF63c9vt5ohbJaU3pp72ukAqXuUdYAItAEkWi6Ejd9_XqG_DUujwxNnyZel6e4bo",
        "title": "",
      },
    },
    {
      "type": "p",
      "value": "====================================================================",
    },
    {
      "type": "h2",
      "value": "An image with a title, description, and link",
    },
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh4.googleusercontent.com/nUTGUYt2OiexwrS7UP6ebivL22RHGBiM6ReeskhJPFHDpcIdw_Piocxg5ccxQBoV-7tc1kS7J9GR7aFml0-k7u4XBwzz1S0IZpdusV_-6vkEN6uM2cCYBSNIZ0boqzaM3IAUF63c9vt5ohbJaU3pp72ukAqXuUdYAItAEkWi6Ejd9_XqG_DUujwxNnyZel6e4bo",
        "title": "",
      },
    },
    {
      "type": "p",
      "value": "====================================================================",
    },
    {
      "type": "h2",
      "value": "An image inserted with a link",
    },
    {
      "type": "p",
      "value": "This image is from https://www.codeforboston.org/img/cfb-logo/cfb-logo-stacked-dark.png",
    },
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh3.googleusercontent.com/QB9v_irPHSgf74e7xTikDiF388boRhrf05fgu7tXBCUyoW1eZyU5Yyy7HNSFNUdLg-sF1fQi3pOtJwFeA0s7cgYs_KOpIK9Eyqdc-9poH_oVxQOANr5GTaGrKzxEXaP-9-u9HO3mkQs6vcPwSMFQkov0ROWUmFW1PIthH6vn3QSnWBaEHWE690xrrauo0VUl2S8",
        "title": "",
      },
    },
    {
      "type": "p",
      "value": "====================================================================",
    },
    {
      "type": "h2",
      "value": "An image inserted with a link and resized",
    },
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh3.googleusercontent.com/1wVP9fuHiumDVk8pWslheSBUJdor8zikP3lTUCs7atcLRdc_fGvFxEp9S3yPyPI-CWTam-777JhCc8mk1MbU8wPfyR09mWrEO3Xb_K0ogc8dlHGS4bXMlGemxrmiBYjgt_V3XEThg3FTdq41Mj9NJm_iO8v3IE3xZElea_n3_7nSGpIRgUUW0L_Ojp1tNOgHRJ4",
        "title": "",
      },
    },
    {
      "type": "p",
      "value": "====================================================================",
    },
    {
      "type": "h2",
      "value": "Centered Image",
    },
    {
      "type": "img",
      "value": {
        "alt": "",
        "source": "https://lh6.googleusercontent.com/CAvp-wuOo2h7nNjyBtIhyyO8Xi4aBI3SngcXmMLj0mQYm-nQSLUedtfKC8DHpZsAl7UGT-JRBY7zpuIBXrrfLrgUziv1NLtnudbKiv0IVux2bdEqSaoquXhjv1jwolGvm-MqlCpC20C5Fsi44vkpKmDHBD6U-sizaFFMaSHfJUVzNaefPAQUvpX-1AihSxFHfY0",
        "title": "",
      },
    },
    {
      "type": "p",
      "value": "====================================================================",
    },
    {
      "type": "h2",
      "value": "Right Justified Image",
    },
  ],
}
`;

exports[`Document "Link" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "p",
      "value": "[Google](https://google.com)",
    },
  ],
}
`;

exports[`Document "Non-continous list" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "ul",
      "value": [
        "",
      ],
    },
    {
      "type": "p",
      "value": "This is a numbered list",
    },
    {
      "type": "ol",
      "value": [
        "Must Have",
      ],
    },
    {
      "type": "p",
      "value": "		Here is an explanation that does not have a number",
    },
    {
      "type": "ol",
      "value": [
        "Should Have Have",
        "Could Have",
        "Won't Have",
      ],
    },
    {
      "type": "p",
      "value": "		Won't have means won't have this release",
    },
    {
      "type": "ol",
      "value": [
        "Really will never have",
        "Really really",
      ],
    },
  ],
}
`;

exports[`Document "Table" to Object 1`] = `
{
  "cover": null,
  "elements": [
    {
      "type": "p",
      "value": "Here is a table",
    },
    {
      "type": "table",
      "value": {
        "headers": [
          "Animal",
          "What I like about the animal",
          "Age",
        ],
        "rows": [
          [
            "Dog",
            "They are man's best friend and great for taking on walks.They are totally accepting",
            "15",
          ],
          [
            "Cat",
            "Meoww!  Need I say more?  Is there any animal with more videos?",
            "12",
          ],
          [
            "Elephant",
            "You know that elephants have tusks but did you know they are dangerous?  In Nepal there are more deaths from elephants than any other animal.  It is against the law to kill an elephant even if it has killed multiple people.",
            "40",
          ],
        ],
      },
    },
    {
      "type": "p",
      "value": "Text in a one cell table will be treated as a code block",
    },
    {
      "type": "table",
      "value": {
        "headers": [
          "if (x = 3) {  print('You are right'}",
        ],
        "rows": [],
      },
    },
    {
      "type": "table",
      "value": {
        "headers": [
          "Fonts",
        ],
        "rows": [
          [
            "**Bold**, _italic_",
          ],
          [
            "**_Bold & Italic_**",
          ],
          [
            "**_<ins>Bold, Italic, and Underline</ins>_**",
          ],
        ],
      },
    },
  ],
}
`;
